### PR TITLE
📖 Remove deprecation message clusterctl

### LINKED ISSUE #2387: [Doc] The deprecation notice of clusterctl in v1alhpa2 doc is out of date
**What steps did you take and what happened:**
CAPI v1alpha2 doc https://cluster-api.sigs.k8s.io/tooling/clusterctl.html says
`This utility has been deprecated in v1alpha2 and will be removed in a future version. `
and https://master.cluster-api.sigs.k8s.io/clusterctl/overview.html (should be the v1alpha3 doc) still provides an updated clusterctl.

Per @detiber's feedback on https://kubernetes.slack.com/archives/C8TSNPY4T/p1582205493288400?thread_ts=1582204591.288200&cid=C8TSNPY4T:

`clusteradm has morphed into clusterctl v2: https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20191016-clusterctl-redesign.md and I believe most (if not all) of the functionality outlined in the proposal should already be merged into the master branch at this point`

**What did you expect to happen:**
Update or Remove the deprecation notice of clusterctl in v1alhpa2 doc https://cluster-api.sigs.k8s.io/tooling/clusterctl.html to comply with the latest clusterctl status.

**Anything else you would like to add:**

**Environment:**
- Cluster-api version: v1alpha2

/kind bug

### PR DESCRIPTION
**Pull request by prankul88**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2387

### CODE REVIEW

File: docs/book/src/tooling/clusterctl.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <aside class="note warning">
-<h1>Deprecation Notice</h1>
+<h1>Notice</h1>
 
-This utility has been deprecated in v1alpha2 and will be removed in a future version. 
+This utility has been updated to `clusterctl v2`. It's design doc can be found [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20191016-clusterctl-redesign.md)
```
Comment (vincepri) [inline docs/book/src/tooling/clusterctl.md]:
```suggestion
This utility, in its current form will be superseded by `clusterctl v2`. A design doc can be found [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20191016-clusterctl-redesign.md)
```

**Comment by vincepri** [comment]
@prankul88 are you interested in changing this message as well? https://github.com/kubernetes-sigs/cluster-api/blob/release-0.2/cmd/clusterctl/cmd/root.go#L29

**Comment by prankul88** [comment]
@vincepri Yup, give me a moment will update. Thanks

File: docs/book/src/tooling/clusterctl.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <aside class="note warning">
-<h1>Deprecation Notice</h1>
+<h1>Notice</h1>
```
Comment (vincepri) [inline docs/book/src/tooling/clusterctl.md]:
```suggestion
<h1>⚠️ Upcoming breaking changes</h1>
```

File: cmd/clusterctl/cmd/root.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 	"k8s.io/klog"
 )
 
-const deprecationMsg string = "NOTICE: clusterctl has been deprecated in v1alpha2 and will be removed in a future version."
+const deprecationMsg string = "NOTICE: This utility, in its current form will be superseded by `clusterctl v2`. A design doc can be found [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20191016-clusterctl-redesign.md)"
```
Comment (vincepri) [inline cmd/clusterctl/cmd/root.go]:
```suggestion
const deprecationMsg string = "NOTICE: This utility, in its current form will be superseded by `clusterctl v2`. For more information see https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20191016-clusterctl-redesign.md"
```

**Comment by ncdc** [comment]
/lgtm

Please close 2387 after this merges as GitHub automation won't do it - thanks!

**Comment by prankul88** [comment]
@vincepri @ncdc Thanks.

**Comment by ncdc** [comment]
@prankul88 you'll need to run `go test ./cmd/clusterctl -update` to get it to update the `cmd/clusterctl/testdata` golden files - that's why the tests are failing.

**Comment by prankul88** [comment]
@ncdc Updated. PTAL

**Pull request merged**